src/solc_0.7/GameToken/GameToken.sol

```diff
@@ -0,0 +1,534 @@
+    function _withdrawalOwnerOf(uint256 id) internal view returns (address) {
+        uint256 data = _owners[_storageId(id)];
+        if ((data & (2**160)) == 2**160) {
+            return address(data);
+        }
+        return address(0);
+    }
+
+    /// @dev Create a new gameId and associate it with an owner.
+    /// @param from The address of one creating the game.
+    /// @param to The address of the Game owner.
+    /// @param subId The id to use when generating the new GameId.
+    /// @param version The version number part of the gameId.
+    /// @param isCreation Whether this is a brand new GAME (as opposed to an update).
+    /// @return id The newly created gameId.
+    function _mintGame(
+        address from,
+        address to,
+        uint64 subId,
+        uint32 version,
+        bool isCreation
+    ) internal returns (uint256 id, uint256 baseId) {
+        uint32 idVersion;
+        uint256 gameId;
+        uint256 storageId;
+        uint256 data;
+        if (isCreation) {
+            idVersion = 1;
+            gameId = _generateGameId(from, subId, idVersion);
+            storageId = _storageId(gameId);
+            data = _owners[storageId];
+            require(data == 0, "BASEID_REUSE_FORBIDDEN");
+        } else {
+            idVersion = version;
+            gameId = _generateGameId(from, subId, idVersion);
+            storageId = _storageId(gameId);
+            data = _owners[storageId];
+        }
+
+        _owners[storageId] = (uint256(idVersion) << 200) + uint256(to);
+
+        emit Transfer(address(0), to, gameId);
+        return (gameId, storageId);
+    }
+
+    /// @dev Allow token owner to set game editors.
+    /// @param gameCreator The address of a GAME creator,
+    /// @param editor The address of the editor to set.
+    /// @param isEditor Add or remove the ability to edit.
+    function _setGameEditor(
+        address gameCreator,
+        address editor,
+        bool isEditor
+    ) internal {
+        emit GameEditorSet(gameCreator, editor, isEditor);
+        _gameEditors[gameCreator][editor] = isEditor;
+    }
+
+    function _bumpGameVersion(address from, uint256 gameId) internal returns (uint256) {
+        address originalCreator = address(gameId / CREATOR_OFFSET_MULTIPLIER);
+        uint64 subId = uint64(gameId / SUBID_MULTIPLIER);
+        uint32 version = uint32(gameId);
+        version++;
+        address owner = _ownerOf(gameId);
+        _burn(from, owner, gameId);
+        (uint256 newId, ) = _mintGame(originalCreator, owner, subId, version, false);
+        address newOwner = _ownerOf(newId);
+        assert(owner == newOwner);
+        return newId;
+    }
+
+    /// @dev Get the baseId (full id without the version number) from the full tokenId.
+    /// @param id The full tokenId for the GAME token.
+    /// @return The baseId.
+    function _storageId(uint256 id) internal pure override returns (uint256) {
+        return uint256(id & BASEID_MASK);
+    }
+
```

Comment:
Let's add a `originalId(uint256 gameId) external view returns (uint256)` function that return the id of the first game token id given an id (it basically return the game token id with version =1)

this can be useful for contract that need to keep track of the lineage